File: phone-app/android/app/src/main/java/com/example/phone_app/background/storage/StorageController.java
package com.example.phone_app.background.storage;

import com.example.phone_app.background.ConnectionService;

import java.util.List;
import java.util.logging.Logger;

import io.objectbox.Box;

public class StorageController {
  private final Box<LocationEntity> box;
  Logger storageLogger = Logger.getLogger(StorageController.class.getName());

  public StorageController(Box<LocationEntity> givenBox) {
    this.box = givenBox;
  }

  public void sendLocationsToServer(long timestamp) {
    removeAllOldLocations(timestamp);
    ConnectionService.sendLocationHistory(box.getAll());
  }

  public void addLocation(Double latitude, Double longitude, Integer givenTimestamp) {
    LocationEntity object = new LocationEntity(latitude, longitude, givenTimestamp);
    box.put(object);
  }

  public List<LocationEntity> genAllOldLocations(Long givenTimestamp) {
    List<LocationEntity> returnedList = null;
    storageLogger.info("Timestamp is " + givenTimestamp.toString());
    returnedList = box.query().less(LocationEntity_.locationTimestamp, givenTimestamp + 1).build().find();

    return returnedList;
  }

  public void removeAllOldLocations(long givenTimestamp) {
    box.remove(genAllOldLocations(givenTimestamp - 7 * 24 * 60 * 60));
  }
}

File: phone-app/ios/Runner/background/storage/StorageController.swift
import Foundation
import ObjectBox

class StorageController {
    private static let box: Box<LocationEntity> = {
        let appSupport = try? FileManager.default.url(for: .applicationSupportDirectory,
                                                     in: .userDomainMask,
                                                     appropriateFor: nil,
                                                     create: true)
                                                     .appendingPathComponent(Bundle.main.bundleIdentifier!)
        let directory = appSupport!.appendingPathComponent("location_history")
        
        try? FileManager.default.createDirectory(at: directory,
                                                 withIntermediateDirectories: true,
                                                 attributes: nil)

        let store = try! Store(directoryPath: directory.path)
        return store.box(for: LocationEntity.self)
    }()
    
    public static func sendLocationsToServer(timestamp: UInt64) {
        self.removeAllOldLocations(givenTimestamp: timestamp)
        ConnectionService.sendLocationHistory(message: try! box.all())
    }
    
    public static func addLocation(latitude: Double, longitude: Double, timestamp: UInt64) {
        let entity:LocationEntity = LocationEntity()
        entity.latitude = latitude
        entity.longitude = longitude
        entity.locationTimestamp = timestamp
        
        try! box.put(entity)
    }
    
    public static func getAllOldLocations(givenTimestamp: UInt64) -> [LocationEntity] {
        NSLog("Timestamp is \(givenTimestamp)")
        let query: Query<LocationEntity> = try! box.query {
            LocationEntity.locationTimestamp.isLessThan(givenTimestamp + 1)
        }.build()
        return try! query.find()
    }
    
    public static func removeAllOldLocations(givenTimestamp: UInt64) {
        try! box.remove(self.getAllOldLocations(givenTimestamp: givenTimestamp - 7 * 24 * 60 * 60))
    }
}
